### ISSUE TITLE
angular_c3_simple not eligible for minification

### ISSUE THREAD

**Issue by divyanshud13** [wasilak/angular-c3-simple#12]
I used angular_c3_simple.js for creating graphs in my rails application. It was working fine in the development environment but it failed on test environment as it was not eligible for minification and was not able to create a minified javascript.

**Comment by wasilak** [comment]
Hi @DivyanshuD ,

Could you provide a little bit more details regarding your minification process?

As you are probably aware, minified version is already provided in `dist` folder of this repository (see: https://github.com/wasilak/angular-c3-simple/blob/master/dist/angular_c3_simple.min.js), so it is possible to do this, but maybe there is something I missed in case of other build systems. I am using `jshint` for this job and unfortunatly I don't work with Rails :/

Peter

**Comment by wasilak** [comment]
Also I after quick look at the code, I think this can be improved to follow Angularjs minification best practices (i.e. see: https://scotch.io/tutorials/declaring-angularjs-modules-for-minification):

``` js
.service('c3SimpleService', function() {
        return {};
    })
```

It should probably look like this:

``` js
.service('c3SimpleService', [function() {
        return {};
    }])
```

I've checked it and it minifies and works correctly afterwards.

If you could test it and confirm if it helps, I can push this to change to repo or accept your pull request.

**Comment by divyanshud13** [comment]
Hi @wasilak 
Thanks for your response. Actually in my project the complete javascript gets minified again and while defining controllers in angular we follow the practice for minification as shown below

 ['$scope',function($scope){
}]

which is not followed in your minified file while declaring controller.

It should be in the form
controller: ['$scope','$element', function($scope, $element) {...

Thanks 
Divyanshu

**Comment by wasilak** [comment]
Hi @DivyanshuD ,

Check out commit c5b30fd7fd52fcbdb706144879a9aab239322b0a. This should address your problem :)

This is included in version `V0.0.12`

**Comment by divyanshud13** [comment]
Hi @wasilak 
The above commit worked out. Thanks :)
